runtime/src/main/java/org/corfudb/protocols/CorfuProtocolLogData.java

```diff
@@ -29,7 +29,8 @@ public static LogDataMsg getLogDataMsg(LogData logData) {
         logData.doSerialize(buf);
 
         return LogDataMsg.newBuilder()
-                .setEntry(ByteString.copyFrom(buf.resetReaderIndex().array()))
+                .setEntry(ByteString.copyFrom(buf.array(), buf.readerIndex(),
+                        buf.writerIndex() - buf.readerIndex()))
```

Comment:
You can just do this `ByteString.copyFrom(buf.array(), buf.readerIndex(), buf.readableBytes()`